homeassistant/components/comelit/config_flow.py

```diff
@@ -0,0 +1,140 @@
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import _LOGGER, DEFAULT_HOST, DEFAULT_PIN, DOMAIN
+
+
+def user_form_schema(user_input: dict[str, Any] | None) -> vol.Schema:
+    """Return user form schema."""
+    user_input = user_input or {}
+    return vol.Schema(
+        {
+            vol.Optional(CONF_HOST, default=DEFAULT_HOST): str,
+            vol.Optional(CONF_PIN, default=DEFAULT_PIN): str,
+        }
+    )
+
+
+STEP_REAUTH_DATA_SCHEMA = vol.Schema({vol.Required(CONF_PIN): str})
+
+
+async def validate_input(
+    hass: core.HomeAssistant, data: dict[str, Any]
+) -> dict[str, str]:
+    """Validate the user input allows us to connect."""
+
+    api = ComeliteSerialBridgeAPi(data[CONF_HOST], data[CONF_PIN])
+
+    try:
+        await api.login()
+    except aiocomelit_exceptions.CannotConnect as err:
+        raise CannotConnect from err
+    except aiocomelit_exceptions.CannotAuthenticate as err:
+        raise InvalidAuth from err
+    finally:
+        await api.logout()
+        await api.close()
+
+    return {"title": data[CONF_HOST]}
+
+
+class ComelitConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Comelit."""
+
+    VERSION = 1
+    entry: ConfigEntry | None = None
+    hassio_discovery: dict[str, Any] | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=user_form_schema(user_input)
+            )
+
+        self._async_abort_entries_match(
+            {CONF_HOST: user_input[CONF_HOST], CONF_PIN: user_input[CONF_PIN]}
+        )
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return self.async_create_entry(title=info["title"], data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=user_form_schema(user_input), errors=errors
+        )
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Handle reauth flow."""
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        assert self.entry
+        self.context["title_placeholders"] = {"host": self.entry.data[CONF_HOST]}
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle reauth confirm."""
+        assert self.entry
```

Comment:
```suggestion
```

Reply:
Need to keep it to make mypy happy on later code, as entry can be None